blog/2023-04-23-json.md

```diff
@@ -0,0 +1,112 @@
+  let id = message.get("id");
+  let content = str.from_json(message.get("content"));
+  b.put("file-${id}.txt", content);
+});
+
+// Test it out
+new cloud.Function(inflight () => {
+  let payload = Json {
+    id: "abc123",
+    content: "Hello, Json!"
+  };
+  some_service.invoke(payload);
+}) as "test: storing messages";
+```
+
+Something noteworthy happens above when we call `str.from_json()`. All fields within a `Json` object are of type `Json`, and Wing does not allow for implicit type conversions from `Json` to other types. Thus, we must explicitly convert the `Json` object to a string using the `from_json()` function, which exists for all Wing primitive types.
+
+Let's dive a little deeper and see what else Wing's `Json` type offers.
+
+### Primitive Types
+
+In line with JSON's ECMA-404 standard, Wing's `Json` type supports the following data primitives:
+
+```js
+let s = Json "string";        // string
+let n = Json 1;               // number
+let b = Json true;            // boolean
+let a = Json [1, 2, 3];       // array
+let o = Json { a: 1, b: 2 };  // object
+```
+
+### Mutability
+
+As with most Wing types, `Json` objects are immutable. This means that once a `Json` object is created, it cannot be modified. However, Wing does provide a way to create a mutable `Json` object. This is done by using the keyword `MutJson` instead of `Json`:
+
+```js
+let j = MutJson { a: 1, b: 2 };
+// We can now modify the json object elsewhere in the code
+j.set("c", 3);
+```
+
+### Heterogeneity
+
+If you have played with Wing much, you're probably aware that nearly all container types in Wing are homogeneous. This is pretty crucial for Wing's type system, but homogeneous data doesn't work well for interoperability between systems. Take the following data for example:
+
+```js
+let id = "abc123";
+let gpa = 3.5;
+let name = "John Doe";
+let completed_courses = ["CS 101", "CS 102", "CS 103"];
+let active = true;
+```
+
+Above, we have some student data, and it's clear that the data is heterogeneous and should be represented as such between systems. This is why Wing's Json type is an exception to the rule of homogeneous containers. The following is a valid Json object in Wing:
+
+```js
+let student = Json {
+  id: "abc123",
+  gpa: 3.5,
+  name: "John Doe",
+  completed_courses: ["CS 101", "CS 102", "CS 103"],
+  active: true
+};
+```
+
+## Mission ~~Accomplished~~ In Progress
+
+While we are excited about the introduction of `Json` into the Wing type system, we are not done yet. There are still a few features we are working on, such as removing the `.get()` and `.set()` syntax for just using the `[]` operator or `.` accessor. We also have some work to do on `Json` schema validation, and some other goodies which can be found in our [language spec](https://docs.winglang.io/reference/spec#114-json-type).
+
+As always, we welcome any feedback and would love to hear from you! Come join our
```

Comment:
Let's decide what's our desired Call To Action for this blogpost and highlight a single CTA - this will allow us to later measure how effective was this blog post.

Reply:
It's OK to have more than one CTA, if we setup our tracking properly we should still be able to measure